Adds Ruby assignment 2 loopings and literals

shirishwebonise opened the pull request.

shirishparsekar committed:
Add ruby assignment 2 - literals and looping

shirishparsekar committed:
Consider first line.. Number of cases

shirishparsekar committed:
Consider first line.. Number of cases

Commit:
Refactor code, Improve loop

Commit:
Refactor code
  - improved readability
  - improved main loop
  - used intersaction of arrays for finding common substring

Commit:
Change 4 space indentation to 2 space indentation

Commit:
Fix tab size to 2 spaces

Commit:
Fix tab size to 2 spaces

Commit:
Fix tab size to 2 spaces

Commit:
Fix indentation

shirishparsekar committed:
Fix naming conventions

shirishparsekar committed:
Add new case

shirishparsekar committed:
Add solution2 using with different approach

shirishparsekar committed:
Add a solution with different approach

shirishparsekar committed:
Remove byebug history

shirishparsekar committed:
Rename variables with ruby conventions

abhayn-webonise commented on Ruby-Assignment-2-LoopingsAndLiterals/Question-1/solution.rb:
```diff
@@ -0,0 +1,44 @@
+input_file = File.open(INPUT_FILE_NAME)
+
+# read number of test cases
+TOTAL_CASES = input_file.readline.chomp.to_i
```
please don't use the variable name in capital letters as in ruby it means it is a constant. 

Have a look http://rubylearning.com/satishtalim/ruby_constants.html

shirishwebonise replied:
Actually I was aware of it, but I thought of making input file name variable to be constant. and similarly total cases variable, since they won't change during the course of the program.

abhayn-webonise replied:
cool :+1:

abhayn-webonise commented on Ruby-Assignment-2-LoopingsAndLiterals/Question-1/solution2.rb:
```diff
@@ -0,0 +1,31 @@
+
+TOTAL_CASES = gets.chomp.to_i
```
please don't use the variable name in capital letters as in ruby it means it is a constant. 

Have a look http://rubylearning.com/satishtalim/ruby_constants.html

shirishwebonise replied:
Actually I was aware of it, but I thought of making it constant since it won't change during the course of the program.

abhayn-webonise commented on Ruby-Assignment-2-LoopingsAndLiterals/Question-1/solution.rb:
```diff
@@ -0,0 +1,44 @@
+
+  # second line - the array elements
+  # read the line, split, convert string elements into integers
+  int_array = input_file.readline.chomp.split.map{ |num| num.to_i }
```
use `input_file.readline.chomp.split.map(&:to_i)` instead

have a look http://stackoverflow.com/questions/1961030/ruby-ampersand-colon-shortcut

abhayn-webonise commented on Ruby-Assignment-2-LoopingsAndLiterals/Question-1/solution.rb:
```diff
@@ -0,0 +1,44 @@
+    left_subarray = int_array[0, index]
+    right_subarray = int_array[index+1, int_array.length]
+
+    if sum_of_numbers(left_subarray) == sum_of_numbers(right_subarray)
```
use guard clause 

```
  return true if sum_of_numbers(left_subarray) == sum_of_numbers(right_subarray)
```

abhayn-webonise commented on Ruby-Assignment-2-LoopingsAndLiterals/Question-1/solution.rb:
```diff
@@ -0,0 +1,44 @@
+
+# method for calculating sum of array elements
+def sum_of_numbers(int_array)
+  total = int_array.reduce(:+)
```
this would also work `int_array.reduce(0, :+)` which would also do the same

abhayn-webonise commented on Ruby-Assignment-2-LoopingsAndLiterals/Question-1/solution2.rb:
```diff
@@ -0,0 +1,31 @@
+
+# method for calculating sum of array elements
+def sum_of_numbers(numbers)
+  total = numbers.reduce(:+)
```
this would also work `int_array.reduce(0, :+)` which would also do the same

shirishwebonise replied:
great.. and since i only have only statement, what if I also remove the function sum_of_numbers and use the statement directly in if condition.

abhayn-webonise replied:
that would be great :+1:

shirishparsekar committed:
Consider impovements
 - map block
 - guard condition
 - shorten reduce statement for summing numbers in array
 - remove sum_of_numbers function and use reduce statement inline instead
